### PR TITLE
[url_launcher] Add null check for extracting browser headers

### PR DESCRIPTION
**Pull request by camsim99**
Adds null check for extracting browser headers.

Fixes https://github.com/flutter/flutter/issues/72863

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/url_launcher/url_launcher/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+## NEXT
+* Fixed extracting browser headers when they are null error.
```
Comment (blasten) [inline packages/url_launcher/url_launcher/CHANGELOG.md:2]:
```suggestion
## NEXT

* Fixed extracting browser headers when they are null error.
```

File: packages/url_launcher/url_launcher/android/src/test/java/io/flutter/plugins/urllauncher/WebViewActivityTest.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import java.util.Map;
+import org.junit.Test;
+
+public class WebViewActivityTest {
+  @Test
+  public void extractHeaders_returnsEmptyMapWhenHeadersBundleNull() {
+    Bundle headersBundle = null;
+
+    Map<String, String> expectedExtractedHeaders = Collections.emptyMap();
+    assertEquals(WebViewActivity.extractHeaders(headersBundle), expectedExtractedHeaders);
```
Comment (blasten) [inline packages/url_launcher/url_launcher/android/src/test/java/io/flutter/plugins/urllauncher/WebViewActivityTest.java:20]:
```suggestion
    assertEquals(WebViewActivity.extractHeaders(null), Collections.emptyMap());
```

**Review by blasten — APPROVED** [review]
LGTM % nits

File: packages/url_launcher/url_launcher/android/src/main/java/io/flutter/plugins/urllauncher/WebViewActivity.java [hunk 3]
```diff
@@ -142,7 +144,10 @@ public void onCreate(Bundle savedInstanceState) {
     registerReceiver(broadcastReceiver, closeIntentFilter);
   }
 
-  private Map<String, String> extractHeaders(Bundle headersBundle) {
+  public static Map<String, String> extractHeaders(@Nullable Bundle headersBundle) {
```
Comment (blasten) [inline packages/url_launcher/url_launcher/android/src/main/java/io/flutter/plugins/urllauncher/WebViewActivity.java:147]: nit: missing annotation: `@visibleForTesting`

File: packages/url_launcher/url_launcher/CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,7 @@
+## NEXT
```
Comment (stuartmorgan) [inline packages/url_launcher/url_launcher/CHANGELOG.md:1]: We should do a version bump on this; we only use `NEXT` when we don't want to publish changes (e.g., test-only changes)

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

**Review by stuartmorgan — APPROVED** [review]
`pubspec.yaml` needs to be updated with the new version (which is why the `publishable` step is failing); otherwise, LGTM. Thank for taking care of this!

**Pull request merged**